### PR TITLE
Feature branch for circleci functionality

### PR DESCRIPTION
**Pull request by aahladmanas**
Circle specific files added
1. circle.yml
2. deploy.sh
3. upload_creds.sh
4. activated_creds.sh

this feature allows for testing and deployment the app.yaml (cron.yaml not yet) to aou-curation-res-test, from the circleci platform. It is able to run bq_utils, spec and validations tests as of now.

### CODE REVIEW

File: data_steward/ci/upload_cloud_creds.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    curl -s -X POST  --header "Content-Type: application/json" \
+        https://circleci.com/api/v1.1/project/$1/$2/$3/envvar?circle-token=$CI_TOKEN \
+        -d "{\"name\": \"$4\", \"value\": \"$5\"}"
+}
+
+GCLOUD_CREDENTIALS_KEY=$(openssl rand -base64 32)
+GCLOUD_CREDENTIALS=$(openssl enc  -aes-256-cbc -in $1 -base64 -A  -k $GCLOUD_CREDENTIALS_KEY)
+
+VCS_TYPE=github
+CI_USERNAME=aahladmanas
```
Comment (mark-velez) [inline data_steward/ci/upload_cloud_creds.sh]: We were supposed to get a system account for this. While I look into that, can you please enable `CI_USERNAME` to be specified by environment variable or script arg? Let me know if you have other ideas.

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    - sudo apt-get update && sudo apt-get install google-cloud-sdk
+    - sudo apt-get install google-cloud-sdk-app-engine-python
+    - chmod 700 ./curation/data_steward/ci/activate_creds.sh
+    - ./curation/data_steward/ci/activate_creds.sh ~/gcloud-credentials-key.json
+
+test:
+  override:
+    - cd ~/curation/data_steward && ./test/run_tests.sh -g /usr/lib/google-cloud-sdk/platform/google_appengine/ -r spec_test.py
+    - cd ~/curation/data_steward && ./test/run_tests.sh -g /usr/lib/google-cloud-sdk/platform/google_appengine/ -r bq_utils_test.py
+      # - cd ~/curation/data_steward && ./test/run_tests.sh -g /usr/lib/google-cloud-sdk/platform/google_appengine/ -r validation_test.py:
```
Comment (mark-velez) [inline circle.yml]: Are these two lines intended to be commented out?

Comment (aahladmanas) [reply circle.yml]: Yet to be decided. I would like to run the tests separately.

File: data_steward/ci/upload_cloud_creds.sh (hunk 1 above)
Comment (aahladmanas) [reply data_steward/ci/upload_cloud_creds.sh]: right. cool

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # All of Us Curation
 
+![alt text](https://circleci.com/gh/aahladmanas/curation/tree/alpha.png?circle-token=f5d4106b20124888e7b626e07ccf31161b6c6ce2 "tests passed")
```
Comment (mark-velez) [inline README.md]: @aahladmanas is there a way to make this associated with the current branch?

**Comment by mark-velez** [comment]
Thanks. This has been merged into the develop branch.

**Pull request closed without merging**